src/main/java/org/apache/sling/dynamicinclude/IncludeTagFilter.java

```diff
@@ -166,8 +167,22 @@ private String getUrl(Configuration config, SlingHttpServletRequest request) {
     private String buildUrl(Configuration config, SlingHttpServletRequest request) {
         final Resource resource = request.getResource();
 
+        // The below code gets the path to the XF and then passes it to the buildUrl method
+        // so that the path to the component is replaced with path to the XF
+        ValueMap vm = resource.adaptTo(ValueMap.class);
```

Comment:
I would base the code on the resource type rather than the existence of a property.

Reply:
Do we need to check for resource type because its already checked before it comes to buildUrl method based on the configuration added in resource-types OSGi config

Reply:
Yes - but this code is going to run for *all* configured resource types.

Reply:
So within the config you can configure multiple resource types and then enable option to rewrite content path instead of component path. So this will be enabled for all configured resource types... 

If there was a 1:1 mapping then we would verify for which resourcetype path rewrite is enabled... Does that make sense?